### PR TITLE
Adding optional pronouns as per #48

### PR DESCRIPTION
**Pull request by harveymatt**
A few things worth mentioning:
- I'm not clear on the intention of a "Speaker" vs their "Profile".
- Preferred gender pronouns are a personal decision so am not adding any validation.
- I found tests using protocols and elements work well for me when starting to deal with combinations of optional fields.

Design:
image

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -7,4 +7,8 @@ target
 *.js
 *.js.map
 *.css
-*.css.map
\ No newline at end of file
+*.css.map
+
+# IDE specific
```
Comment (harveymatt) [inline .gitignore:12]: Would you normally put this in a separate PR or are small changes like this allowed along side a feature PR?

File: client/src/main/scala/lsug/ui/event/Speaker.scala [hunk 2]
```diff
@@ -67,7 +71,7 @@ object Speaker {
           speaker
             .map {
               case P.Speaker(
-                  p @ P.Speaker.Profile(_, name, _),
+                  p @ P.Speaker.Profile(_, name, _, _),
```
Comment (harveymatt) [inline client/src/main/scala/lsug/ui/event/Speaker.scala:74]: I used a placeholder to avoid naming collisions but am happy to change

File: protocol/src/main/scala/lsug/protocol/protocol.scala [hunk 3]
```diff
@@ -216,14 +226,15 @@ object Speaker {
...
+      photo: Option[Asset],
+      pronoun: Option[Pronoun]
   )
 
   object Profile {
     implicit val codec: Codec[Profile] = deriveCodec[Profile]
     implicit val eq: Eq[Profile] = Eq.instance {
-      case (Profile(i, n, p), Profile(ii, nn, pp)) =>
-        i === ii && n === nn && p === pp
+      case (Profile(i, n, p, q), Profile(ii, nn, pp, qq)) =>
```
Comment (harveymatt) [inline protocol/src/main/scala/lsug/protocol/protocol.scala:236]: pronoun obviously doesn't start with `q` but `p` was already taken here - what would your convention be instead?

**Review by harveymatt — COMMENTED** [review]
Added a few queries along side my PR - hope you don't mind

File: .gitignore (hunk 1 above)
Comment (zainab-ali) [reply .gitignore:12]: Here is absolutely fine :+1:

File: client/src/main/scala/lsug/ui/event/Speaker.scala [hunk 4]
```diff
@@ -58,6 +58,9 @@ object Speaker {
       }
       .getOrElse(ProfilePicture(profile.some): TagMod)
 
+  private def pronoun(profile: P.Speaker.Profile) =
+    profile.pronoun.map { p => <.h4(s"${p.show}") }
```
Comment (zainab-ali) [inline client/src/main/scala/lsug/ui/event/Speaker.scala:62]: A heading implies a semantic section of text.  This would be better placed in a paragraph, or a span with appropriate css styles:
```html
<p>Referred to as <strong>he/him</strong></p>
```
It should’t be a `h4` as this causes problems for screen readers. A screen reader may read the bio as being under the he/him heading.

**Comment by zainab-ali** [comment]
Thank you very much for making this!  It's a great start.  I've a few suggestions on the design front.

# The speaker vs the speaker’s profile
The speaker profile refers to the summary on the main https://www.lsug.co.uk/ page. This has a summary of the past events. Each speaker has a name and photo, but nothing more (the `Speaker.Profile`).
You can click on a summary to go to a detailed page https://www.lsug.co.uk/events/2020-11-28/8. This has all the speaker information (the `Speaker`).

In terms of http request payloads; on the main page, we only send the profile. On the detail page, we send all the information.

Let me know if this isn't clear and I'll dig out some screenshots.

# Displaying the pronoun
Attendees have a wide net of cultural backgrounds, and some won’t know the significance of gender pronouns. Nevertheless, they’ll be enforced at meetups (attendees will be out of line if they refuse to use the pronoun listed). We should emphasize the pronoun, and perhaps give some context with “referred to as”.  What do you think of this design?

PHOTO TO BE INSERTED

# Hard-coding a list of pronouns
Gender pronouns are definitely up to a speaker to decide.  However, we’d like to standardize their format.
For example, `he/him` as opposed to `he,him` or `him,he` are all different formats of the same pronoun.
My preference is lower-cased, with a ‘/’.

We can achieve this by hard coding a list of pronouns (perhaps in the `Decoders.scala`), and looking up the corresponding text.

``` scala
final case class Pronoun(subject: String, obj: String)

...

val recognizedPronouns: List[Pronoun] = List(
  Pronoun("he", "him"),
  Pronoun("she", "her"),
  Pronoun("they", "them"))

def pronoun(text: String): Either[Error, Pronoun] = {

  // split the text by a / and look up the pronoun in the list
  ...
}
```

We’ll encourage speakers to edit this list if their pronoun isn’t present (we'll mention it when requesting the speaker for content).


# To summarize
First, think about whether you agree with the design decisions:
 - the pronoun shouldn’t be on the main page
 - the pronoun should be under the speaker picture, with the text “referred to as …”

If not, please feel free to discuss them.  I’m no expert in design, and there could be things I haven’t thought through.

If you’re happy with this:
 - Move the pronoun field from the `SpeakerProfile` to the `Speaker`.  This is because we're displaying it on the detail page, not the main page
 - Add the list of recognized pronouns as above
 - Edit the `people-template.pm` and possibly `speaker-details.md` issue template to include pronoun guidance
 
Please feel free to discuss this in the Gitter chat.  Thanks again!

File: protocol/src/main/scala/lsug/protocol/protocol.scala (hunk 3 above)
Comment (zainab-ali) [reply protocol/src/main/scala/lsug/protocol/protocol.scala:235]: I'm not particularly fussy with these things.  Personally, I tend to use full words for anything with meaning, and keep to letters for abstract types.
For example, this would always be `fa` and `f`, because there's nothing more concrete it could mean:

```scala
def map(fa: F[A])(f: A => B): F[B] = ...
```

I'd also use a short form if the type was clearly inferred, like:
```scala
val ns: List[SpeakerName] = ???
```

But if the type was ambiguous, I'd use a full word:
```scala
val speakerNames: List[String] = ???
```

This is entirely a matter of personal preference as everyone finds different things easier to read.  There's no fixed convention around it.

In this case, what you have is pretty clear.

**Review by zainab-ali — CHANGES_REQUESTED** [review]

**Comment by zainab-ali** [comment]
Hi @harveymatt , would it be ok if I shepherded the rest of this through myself? We have a speaker this Saturday with a they/them pronoun, and it would be nice to accommodate them properly.

**Review by zainab-ali — APPROVED** [review]

**Comment by zainab-ali** [comment]
Thanks @harveymatt ! I made a few last tweaks and pushed this through.
